examples/dynamic-forall.cpp

```diff
@@ -77,28 +77,25 @@ int main(int argc, char *argv[])
 
 //----------------------------------------------------------------------------//
 
-  std::cout << "\n Running C-style vector addition...\n";
-
-  // _cstyle_vector_add_start
-  for (int i = 0; i < N; ++i) {
-    c[i] = a[i] + b[i];
-  }
-  // _cstyle_vector_add_end
-
-  checkResult(c, N);
-//printResult(c, N);
-
-
-//----------------------------------------------------------------------------//
-// Example of dynamic policy selection for forall
-//----------------------------------------------------------------------------//
+  std::cout << "\n Running dynamic forall vector addition and reductions...\n";
 
+  int sum = 0;
+  using VAL_INT_SUM = RAJA::expt::ValOp<int, RAJA::operators::plus>;
+  
+  RAJA::RangeSegment range(0, N); 
+  
   //policy is chosen from the list
-  RAJA::expt::dynamic_forall<policy_list>(pol, RAJA::RangeSegment(0, N), [=] RAJA_HOST_DEVICE (int i)   {
+  RAJA::dynamic_forall<policy_list>(pol, range,
+    RAJA::expt::Reduce<RAJA::operators::plus>(&sum),
+      RAJA::expt::KernelName("RAJA dynamic forall"),
```

Comment:
Should we pull off more of the band-aid and move `KernelName` out of the `expt` namespace?

Reply:
`KernelName` functions within the new reducer framework, so I think we need to keep it in `expt` until the new reducers are fully integrated.

Reply:
It seems like it should work in kernels without reductions also. Maybe that's a topic for a later discussion.

Reply:
I'm using the KernelName with the RAJA caliper integration here: https://github.com/LLNL/RAJA/pull/1773. I would be in favor in getting it out of expt to prompt folks to start using it more confidently.

Reply:
Thats a good point, I agree with @trws. It should stay in expt until we move the new reducer interface out of expt.